cdap-ui/app/cdap/components/FastAction/SetPreferenceAction/SetPreferenceModal.js

```diff
@@ -405,14 +448,35 @@ export default class SetPreferenceModal extends Component {
                     <button
                       className="btn btn-primary float-xs-left"
                       onClick={this.setPreferences}
-                      disabled={(!this.allFieldsFilled() || this.state.error) ? 'disabled' : null}
+                      disabled={(this.oneFieldMissing() || this.state.error) ? 'disabled' : null}
                     >
                       <span>{saveAndCloseLabel}</span>
                     </button>
                 }
                 <span className="float-xs-left reset">
                   <a onClick = {this.resetFields}>{resetLink}</a>
                 </span>
+                {
+                  this.state.showResetMessage ?
+                    (
+                      <span className="float-xs-left text-success reset-success">
+                        Reset Successful
+                      </span>
+                    )
+                  :
+                    null
+                }
+                <span className="float-xs-right num-rows">
+                  {
+                    this.state.keyValues.pairs ?
+                      this.state.keyValues.pairs.length === 1 ?
```

Comment:
It would be nice if we don't nest ternary operators. It becomes a little hard to read.